src/parseTools.js

```diff
@@ -1291,3 +1291,17 @@ function makeMalloc(source, param) {
   }
   return `abort('malloc was not included, but is needed in ${source}. Adding "_malloc" to EXPORTED_FUNCTIONS should fix that. This may be a bug in the compiler, please file an issue.');`;
 }
+
+// Adds a call to runtimeKeepalivePush, but not under MINIMAL_RUNTIME.
+// Avoids having many `if MINIMAL_RUNTIME` sections in the library code.
+function runtimeKeepalivePush() {
+  if (!MINIMAL_RUNTIME) return '';
+  return 'runtimeKeepalivePush();';
+}
+
+// Adds a call to runtimeKeepalivePop, but not under MINIMAL_RUNTIME.
+// Avoids having many `if MINIMAL_RUNTIME` sections in the library code.
+function runtimeKeepalivePop() {
+  if (!MINIMAL_RUNTIME) return '';
+  return 'runtimeKeepalivePop();';
+}
```

Comment:
Can we also avoid the code size in the case of `EXIT_RUNTIME=0`?

I realize it is more complicated for pthreads builds but in single-threaded code that will never exit the runtime - the default build - it seems like we never exit and so we do not need to track liveness of the runtime?

Reply:
Ooh, good idea.   I'll try that.

Reply:
Done!